src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.spec.js

```diff
@@ -1,283 +1,272 @@
-  selectedLanguage: 'en',
-  intl: { formatMessage: message => message.id },
-  ...custom,
-});
+// We use this component to simulate the whole flow of
+// changing a value and formatting on blur.
+class TestComponent extends React.Component {
+  static displayName = 'TestComponent';
+  static propTypes = {
+    id: PropTypes.string,
+    value: PropTypes.objectOf(PropTypes.string).isRequired,
+    onChange: PropTypes.func,
+    selectedLanguage: PropTypes.string.isRequired,
+  };
+  static defaultProps = {
+    id: 'some-id',
+    name: 'some-name',
+    value: {
+      en: 'Oh no, oh no, oh no. no',
+      fr: "Calisse d'ostie de tabarnak",
+    },
+    selectedLanguage: 'en',
+    intl: { formatMessage: message => message.id },
+  };
 
-describe('rendering', () => {
-  let wrapper;
-  let props;
-  describe('with minimal props', () => {
-    beforeEach(() => {
-      props = createTestProps();
-      wrapper = shallow(<LocalizedMultilineTextInput {...props} />);
-    });
-    it('should match snapshot', () => {
-      expect(wrapper).toMatchSnapshot();
+  state = {
+    value: this.props.value || {},
+    selectedLanguage: this.props.selectedLanguage || '',
+  };
+
+  handleChange = event => {
+    event.persist();
+    this.setState({
+      value: {
+        ...this.state.value,
+        [event.target.language]: event.target.value,
+      },
     });
+  };
+
+  render() {
+    return (
+      <LocalizedMultilineTextInput
+        {...this.props}
+        onChange={this.props.onChange || this.handleChange}
+        value={this.state.value}
+        selectedLanguage={this.state.selectedLanguage}
+      />
+    );
+  }
+}
+
+const renderLocalizedMultilineTextInput = (props, options) =>
+  render(<TestComponent {...props} />, options);
+
+it('should forward data-attributes', () => {
+  const { container } = renderLocalizedMultilineTextInput({
+    'data-foo': 'bar',
   });
-  describe('when collapsed', () => {
-    describe('with id', () => {
-      beforeEach(() => {
-        props = createTestProps({ id: 'foo' });
-        wrapper = shallow(<LocalizedMultilineTextInput {...props} />);
-      });
-      it('should add a language suffix to the id of each visible input', () => {
-        // only the first language input is visible here
-        expect(wrapper).toRender({ id: 'foo.en' });
-      });
-    });
-    describe('with name', () => {
-      beforeEach(() => {
-        props = createTestProps({ name: 'foo' });
-        wrapper = shallow(<LocalizedMultilineTextInput {...props} />);
-      });
-      it('should forward the name', () => {
-        // only the first language input is visible here
-        expect(wrapper).toRender({ name: 'foo.en' });
-      });
-    });
+  expect(container.querySelector('[data-foo="bar"]')).toBeInTheDocument();
+});
+
+it('should have an HTML name', () => {
+  const { container } = renderLocalizedMultilineTextInput({ name: 'foo' });
+  expect(
+    container.querySelector(`[name="${getName('foo', 'en')}"]`)
```

Comment:
No label?

Reply:
[484a79e3dbafac9f4779aad77ad3bff779cb76f9](https://github.com/commercetools/ui-kit/pull/314/commits/484a79e3dbafac9f4779aad77ad3bff779cb76f9) is better?